app/src/main/java/com/babylon/wallet/android/domain/usecases/transaction/PollTransactionStatusUseCase.kt

```diff
@@ -1,106 +1,68 @@
 package com.babylon.wallet.android.domain.usecases.transaction
 
 import com.babylon.wallet.android.data.dapp.model.TransactionType
-import com.babylon.wallet.android.data.gateway.generated.models.TransactionPayloadStatus
 import com.babylon.wallet.android.data.repository.TransactionStatusData
 import com.babylon.wallet.android.data.repository.transaction.TransactionRepository
 import com.babylon.wallet.android.domain.RadixWalletException
 import com.radixdlt.sargon.Epoch
-import kotlinx.coroutines.delay
+import com.radixdlt.sargon.IntentHash
+import com.radixdlt.sargon.TransactionStatus
+import com.radixdlt.sargon.TransactionStatusReason
 import javax.inject.Inject
 
+private const val EPOCH_WINDOW_DURATION_IN_MINUTES = 5
```

Comment:
The logic of converting epoch in aprox minutes is also used in AssetDetails dialog when previewing a stake unit that will be ready for claim in X minutes. Could we instead create a function that converts epochs to minutes with a parameter on how much time an epoch takes which will default in 5min?